core/templates/dev/head/player/PlayerServices.js

```diff
@@ -489,6 +495,63 @@ oppia.controller('LearnerLocalNav', [
 
   $scope.showEmbedExplorationModal = embedExplorationButtonService.showModal;
 
+  $scope.showLearnerSuggestionModal = function() {
+    $modal.open({
+      templateUrl: 'modals/learnerViewSuggestion',
+      backdrop: true,
+      resolve: {},
+      controller: [
+        '$scope', '$modalInstance', '$timeout', 'playerPositionService',
+        'oppiaPlayerService',
+        function($scope, $modalInstance, $timeout, playerPositionService,
+                 oppiaPlayerService) {
+          var stateName = playerPositionService.getCurrentStateName();
+          $scope.initContent = oppiaPlayerService.getStateContentHtml(stateName);
+          $scope.suggestionContent = $scope.initContent;
+          $scope.showEditor = false;
+          // Rte initially displays content unrendered for a split second
+          $timeout(function() {
+            $scope.showEditor = true;
+          }, 500);
+
+          $scope.cancelSuggestion = function() {
+            $modalInstance.dismiss('cancel');
+          };
+
+          $scope.submitSuggestion = function() {
+            $modalInstance.close({
+              data: {
+                id: oppiaPlayerService.getExplorationId(),
+                version: oppiaPlayerService.getExplorationVersion(),
+                stateName: stateName,
+                suggestionContent: $scope.suggestionContent
+              }
+            });
+          };
+        }]
+    }).result.then(function(result) {
+      $http.post('/suggestionhandler/' + result.data.id, {
+        exploration_version: result.data.version,
+        state_name: result.data.stateName,
+        suggestion_content: {
+          type: 'text',
+          value: result.data.suggestionContent
+        }
+      }).success(function(res) {
+        $modal.open({
```

Comment:
Can we do this immediately rather than waiting for the post() to return? There's an awkward pause after the submission but before the follow-up modal opens.

Please also check what would happen if there was some kind of server error -- a yellow warning message should pop up. (I.e. there should be some kind of error handler.)